src/main/java/org/wildfly/client/config/ClientConfiguration.java

```diff
@@ -67,6 +68,10 @@
     }
 
     private InputStream streamOpener() throws IOException {
+        if (configurationUri == null) {
+            throw new FileNotFoundException("Configuration file is not loaded");
```

Comment:
This would need to be i18n... however I think maybe it's more correct to null-check the parameter in the `getInstance(...)` methods.

Reply:
Did you look into null-checking the parameter?